internal/command/console/console.go

```diff
@@ -210,15 +233,29 @@ func makeEphemeralConsoleMachine(ctx context.Context, app *api.AppCompact, appCo
 	if err != nil {
 		return nil, nil, err
 	}
-	if currentRelease == nil {
+	if currentRelease == nil && !flag.IsSpecified(ctx, "image") {
 		return nil, nil, errors.New("can't create an ephemeral console machine since the app has not yet been released")
 	}
 
 	machConfig, err := appConfig.ToConsoleMachineConfig()
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to generate ephemeral console machine configuration: %w", err)
 	}
-	machConfig.Image = currentRelease.ImageRef
+
+	if flag.IsSpecified(ctx, "image") {
+		machConfig.Image = flag.GetString(ctx, "image")
+	} else {
+		machConfig.Image = currentRelease.ImageRef
+	}
+
+	if env := flag.GetStringArray(ctx, "env"); len(env) > 0 {
+		parsedEnv, err := cmdutil.ParseKVStringsToMap(env)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed parsing environment: %w", err)
+		}
+		machConfig.Env = parsedEnv
```

Comment:
In the [`ToConsoleMachineConfig`](https://github.com/superfly/flyctl/blob/4e09fc60ad452dda368372a9767386d8a5c0549d/internal/appconfig/machines.go#L64) call, `machConfig.Env` is populated with environment variables set in the app's configuration and a few special ones, e.g. `FLY_PROCESS_GROUP`.

In light of this, I think you mean to add/overwrite the individual variables, rather than to replace the whole map?

Reply:
Good catch!  In my case `FLY_PROCESS_GROUP` and `PRIMARY_REGION` were stomped on.  Fixed.